bundles/config/org.eclipse.smarthome.config.discovery.mdns/src/main/java/org/eclipse/smarthome/config/discovery/mdns/internal/MDNSDiscoveryService.java

```diff
@@ -147,14 +152,24 @@ private void scan() {
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
     protected void addMDNSDiscoveryParticipant(MDNSDiscoveryParticipant participant) {
+        if (participant instanceof ExtendedDiscoveryService) {
+            ((ExtendedDiscoveryService) participant).setDiscoveryServiceCallback(discoveryServiceCallback);
```

Comment:
I doubt that this will work: when the participants are added to the MDNSDiscoveryService the `discoveryServiceCallback` is not necessarily set.

Reply:
That is true, but therefore I have the loop over all participants when the callback is set.

This code here is for the case that **later** at runtime a participant is added, so it receives the callback as well.